### PR TITLE
Simplify Packet Broker settings

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

Closes [#3838](https://github.com/TheThingsIndustries/lorawan-stack/issues/3838)

#### Changes

- Update packet broker main view
- Separate two forms
- Add radio button options
- Update e2e tests
- Remove gateway visibility

Screenshot 2023-10-18 at 14-47-15 Management platform for The Things Stack for LoRaWAN

#### Testing

1. Open packet broker in the admin panel.
2. Select the first two options one at a time to make sure each one is successful.
3. Select custom configuration, go to the network tab and change a network config.
4. Reload the page see that the configuration you selected is the one shown.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] The steps/process to test this feature are clearly explained including testing for regressions.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by johanstokking — CHANGES_REQUESTED** [review]
I tested this locally and I have a few issues:

1. If you select sandbox, it creates a policy with NetID `000013` but not tenant ID `ttn`. It should be with tenant ID `ttn` explicitly because that is the community network.
2. When I go from the first radio button (forward traffic to all networks) and save, there's indeed a default routing policy created. However, the home network routing policy with `00013` is still there. I think all other policies should be deleted
3. When I go back to the second radio button (forward to sandbox) and click save, I get the error [0]

[0]

```
{
  "code": 3,
  "message": "type mismatch, parameter: net_id, error: strconv.ParseUint: parsing \"NaN\": invalid syntax"
}
```

**Comment by ryaplots** [comment]
> I tested this locally and I have a few issues:
> 
>     1. If you select sandbox, it creates a policy with NetID `000013` but not tenant ID `ttn`. It should be with tenant ID `ttn` explicitly because that is the community network.
> 
>     2. When I go from the first radio button (forward traffic to all networks) and save, there's indeed a default routing policy created. However, the home network routing policy with `00013` is still there. I think all other policies should be deleted
> 
>     3. When I go back to the second radio button (forward to sandbox) and click save, I get the error [0]
> 
> 
> [0]
> 
> ```
> {
>   "code": 3,
>   "message": "type mismatch, parameter: net_id, error: strconv.ParseUint: parsing \"NaN\": invalid syntax"
> }
> ```

1. When trying to create a policy with an explicit tenant id (this url: `https://tti.staging1.cloud.thethings.industries/api/v3/pba/home-networks/policies/19/ttn`) I get an error:
```
{
  "code": 10,
  "message": "error:unknown:unknown (bad reference)",
  "details": [
    {
      "@type": "type.googleapis.com/ttn.lorawan.v3.ErrorDetails",
      "message_format": "bad reference",
      "correlation_id": "fa423e90d1854199bb417f4497ffb3be",
      "code": 10
    }
  ]
}
```
I checked the [staging networks list](https://tti.staging1.cloud.thethings.industries/console/admin-panel/packet-broker/networks) and it seems like The Things Network network does not have a tenant id. Also when updating its policy the request is sent to `https://tti.staging1.cloud.thethings.industries/api/v3/pba/home-networks/policies/19`, which is why I used the same net_id and did not indicate a tenant ID.

2. I will update this.

3. Also looking into fixing this.

@johanstokking

**Comment by kschiffer** [comment]
> If you select sandbox, it creates a policy with NetID 000013 but not tenant ID ttn. It should be with tenant ID ttn explicitly because that is the community network.

TTN is not on a tenant, is it? It's a single-tenant network, or what am I missing?

File: pkg/webui/console/components/default-routing-policy-form/index.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
```
Comment (kschiffer) [inline pkg/webui/console/components/default-routing-policy-form/index.js]:
```suggestion
// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
```

File: pkg/webui/console/components/default-routing-policy-form/routing-policy-form.styl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+.direction-header
+  margin-bottom: $cs.xs
+
+.policy-source
+  margin-bottom: $ls.s
```
Comment (kschiffer) [inline pkg/webui/console/components/default-routing-policy-form/routing-policy-form.styl]: We don't really need these anymore, can be replaced by utility classes.

File: pkg/webui/console/components/payload-formatters-form/index.js [hunk 3]
```diff
@@ -419,7 +418,7 @@ const PayloadFormattersForm = ({
           >
             {() => (
               <>
-                <Form.SubTitle title={m.setupSubTitle} />
+                <Form.SubTitle title={sharedMessages.setupSubTitle} />
```
Comment (kschiffer) [inline pkg/webui/console/components/payload-formatters-form/index.js]: Please replace the name of this message to just `setup`.

File: pkg/webui/console/components/routing-policy-form/index.js [hunk 4]
```diff
@@ -39,7 +39,6 @@ const m = defineMessages({
   saveDefaultPolicy: 'Save default policy',
   useDefaultPolicy: 'Use default routing policy for this network',
   useSpecificPolicy: 'Use network specific routing policy',
-  doNotUseADefaultPolicy: 'Do not use a default routing policy for this network',
```
Comment (kschiffer) [inline pkg/webui/console/components/routing-policy-form/index.js:42]: Makes me realize that we don't offer deleting the default policy anymore with this change. What's the difference from setting an all-zero default routing policy @johanstokking ?

Asking to understand whether we need to still include the option of deleting the default routing policy somehow.

File: pkg/webui/console/views/admin-packet-broker/admin-packet-broker.js [hunk 5]
```diff
@@ -126,7 +177,79 @@ const PacketBroker = () => {
     },
   ]
 
-  const boldMessage = { b: msg => <b>{msg}</b> }
+  const defaultRoutingPolicy = useSelector(selectHomeNetworkDefaultRoutingPolicy)
+  const routingPolicies = useSelector(selectPacketBrokerHomeNetworkPoliciesStore)
+  const initialValues = {
+    _routing_configuration:
+      peerWithEveryNetwork(defaultRoutingPolicy) && isValidPolicy(defaultRoutingPolicy)
```
Comment (kschiffer) [inline pkg/webui/console/views/admin-packet-broker/admin-packet-broker.js]: This is not quite enough. You also need to check that no individual network has any custom routing policy set. Right now, if you for example manually set a fully permissive routing policy but then override this policy with a custom one for an individual network, the UI will still show "Enable peering with all networks" even though it is still a custom configuration.

```js
peerWithEveryNetwork(defaultRoutingPolicy) && routingPolicies.length === 0 && isValidPolicy(defaultRoutingPolicy)
```

File: pkg/webui/console/views/admin-packet-broker/default-gateway-visibility.js [hunk 6]
```diff
@@ -1,73 +0,0 @@
-// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
```
Comment (kschiffer) [inline pkg/webui/console/views/admin-packet-broker/default-gateway-visibility.js]: The gateway visibility tab actually needs to stay. This accidentally went missing in the wireframe.

The way this should work now is using a top level "Routing configuration"-tab and a "Default gateway visibility"-tab.

**Comment by johanstokking** [comment]
> > If you select sandbox, it creates a policy with NetID 000013 but not tenant ID ttn. It should be with tenant ID ttn explicitly because that is the community network.
> 
> TTN is not on a tenant, is it? It's a single-tenant network, or what am I missing?

Yeah it actually is a tenant, but not on The Things Stack Cloud. The Things Stack Sandbox is a single tenant deployment that uses NetID `000013` and tenant ID `ttn`. For Packet Broker, there's no difference between `ttn` and TTSC tenants.

@ryaplots indeed the `ttn` tenant was missing on Packet Broker staging. It is there now.

**Review by kschiffer — CHANGES_REQUESTED** [review]
The default gateway visibility configuration should not be deleted but moved to a separate tab next to "Routing configuration".

**Comment by kschiffer** [comment]
See this updated wireframe:
image

File: pkg/webui/console/components/routing-policy-form/index.js (hunk 4 above)
Comment (johanstokking) [reply pkg/webui/console/components/routing-policy-form/index.js:42]: The difference between no default routing policy and an all-zero routing policy is that with no default policy, there's a fallback to an underlying default policy.

For example, on the NetID level, there can be a default policy. When tenants do not have a default policy, they fallback to the NetID's default policy. When the tenant has a default policy that is empty, there's no fallback.

Comment (ryaplots) [reply pkg/webui/console/components/routing-policy-form/index.js:42]: So we do still need the `'Do not use a default routing policy for this network'` option, right? @kschiffer

**Comment by ryaplots** [comment]
> See this updated wireframe:

Where can I get the texts from the tooltips of the routing options? @kschiffer

**Comment by kschiffer** [comment]
Let's skip them for now.

**Comment by kschiffer** [comment]
We need to update the routing there in general a bit and make `/routing-configuration` an explicit routes, otherwise there is ambiguity in the active link selection. (Try selecting the network tab within routing configuration and notice that `Routing configuration` is not selected anymore.

I've taken a shot at this and added a fix commit.

**Review by kschiffer — APPROVED** [review]
Please do a final test of the new code before merging.

**Comment by johanstokking** [comment]
1. What I'm now missing is a way to reset the default routing policy. If you now have custom selected, you have an empty default routing policy, even if none is set. If you click Save, it saves an empty default routing policy. But as outlined in https://github.com/TheThingsNetwork/lorawan-stack/pull/6639#discussion_r1369849111, there is a difference between no default routing policy and an all-zero default routing policy.

2. If you have "Forward traffic to all networks registered in Packet Broker" selected (and saved) and then click Custom, it would be nice to have the default routing policy enabled and all-checked, because that's what the current state is.

File: pkg/webui/console/components/routing-policy-form/index.js (hunk 4 above)
Comment (kschiffer) [reply pkg/webui/console/components/routing-policy-form/index.js:42]: Missed that. Yes indeed, we need these two options `Do not use a default routing policy for this network` and `Use a default routing policy`.

For added clarity, we'll probably want to label this  `Do not set a default routing policy (use global network default)`.

@johanstokking do you think that's accurate enough?

Comment (johanstokking) [reply pkg/webui/console/components/routing-policy-form/index.js:42]: Yes that sounds good

**Comment by ryaplots** [comment]
> 1. What I'm now missing is a way to reset the default routing policy. If you now have custom selected, you have an empty default routing policy, even if none is set. If you click Save, it saves an empty default routing policy. But as outlined in [Simplify Packet Broker settings #6639 (comment)](https://github.com/TheThingsNetwork/lorawan-stack/pull/6639#discussion_r1369849111), there is a difference between no default routing policy and an all-zero default routing policy.
> 
>     2. If you have "Forward traffic to all networks registered in Packet Broker" selected (and saved) and then click Custom, it would be nice to have the default routing policy enabled and all-checked, because that's what the current state is.

I have taken this into account and added back the functionality. @johanstokking Please review.

**Review by johanstokking — APPROVED** [review]
Thanks, this now works as expected.

**Pull request merged**